src/components/Map.jsx

```diff
@@ -1,185 +1,185 @@
-};
+const apiKey = import.meta.env.VITE_GOOGLE_MAPS_API_KEY;
+// Define the libraries required for Google Maps
+const LIBRARIES = ['marker'];
 
-const userIcon = L.divIcon({
-  className: '',
-  html: `<div class="user-location-circle"></div>`,
-  iconSize: [20, 20],
-  iconAnchor: [10, 10],
-});
-
-const getParkingIcon = (availableSpaces) => {
-  const iconUrl = availableSpaces > 0 ? availableIcon : fullIcon;
-  return L.icon({
-    iconUrl,
-    iconSize: [35, 45],
-    iconAnchor: [12, 41],
-    popupAnchor: [0, -41],
+// Component for rendering the map with parking spots
+const Map = memo(({ parkingSpots, selectedSpot, setSelectedSpot }) => {
+  const [userLocation, setUserLocation] = useState(null);
+  const [infoWindowOpen, setInfoWindowOpen] = useState(false);
+  const mapRef = useRef(null);
+  const markersRef = useRef([]);
+  const userCircleRef = useRef(null);
+
+  // Load the Google Maps API
+  const { isLoaded, loadError } = useJsApiLoader({
+    googleMapsApiKey: apiKey,
+    libraries: LIBRARIES,
   });
-};
 
-const ZoomToUserLocation = ({ userLocation }) => {
-  const map = useMap();
-
-  useEffect(() => {
-    if (userLocation) {
-      map.setView(userLocation, 16);
+  // Ubicar al usuario y centrar el mapa en su posición
+  const zoomToUserLocation = useCallback((location) => {
+    if (mapRef.current) {
+      mapRef.current.setCenter(location);
+      mapRef.current.setZoom(16); // Nivel de zoom al usuario
     }
-  }, [userLocation, map]);
-
-  return null;
-};
+  }, []);
 
-export function Map({ onParkingSpotSelected }) {
-  const [parkingSpots, setParkingSpots] = useState([]);
-  const [userLocation, setUserLocation] = useState(null);
-  const circleRadius = 20;
-
-  const fetchSpots = useCallback(async () => {
-    try {
-      const data = await fetchParkingSpots();
-      const uniqueData = data.filter(
-        (value, index, self) =>
-          index === self.findIndex((t) => t.id === value.id),
-      );
-      setParkingSpots(uniqueData);
-    } catch (error) {
-      console.error('Error al obtener los parqueaderos:', error);
-    }
+  // Locate the user and center the map on their position
+  const locateUser = useCallback(() => {
+    navigator.geolocation?.getCurrentPosition(
+      ({ coords: { latitude, longitude } }) => {
+        const newLocation = { lat: latitude, lng: longitude };
+        setUserLocation(newLocation);
+        zoomToUserLocation(newLocation);
+        mapRef.current?.panTo(newLocation);
+
+        if (userCircleRef.current) userCircleRef.current.setMap(null);
+        userCircleRef.current = new window.google.maps.Circle({
+          map: mapRef.current,
+          center: newLocation,
+          radius: 30,
+          strokeColor: '#4285F4',
+          fillColor: '#4285F4',
+          fillOpacity: 0.35,
+        });
+      },
+      (error) => console.error('Error fetching location:', error),
```

Comment:
I saw you handling errors with Spanish messages above. I'm fine with either language to send error messages to the user, Let's keep one single language for it.